src/zone/worker.cpp

```diff
@@ -48,6 +48,9 @@ struct Worker::Impl {
 
     /// <summary> A callback function that is called when worker becomes idle. </summary>
     std::function<void(WorkerId)> idleNotificationCallback;
+
+    /// <summary> The zone settings for the current worker. </summary>
+    const settings::ZoneSettings* settings;
```

Comment:
Please consider the life cycle of this pointer, since you created a 'start' member function which will refer to it.

Reply:
We should make a copy of ZoneSettings